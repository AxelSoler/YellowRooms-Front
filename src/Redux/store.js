import {
  legacy_createStore as createStore,
  combineReducers,
  applyMiddleware,
} from "redux";
import thunk from "redux-thunk";
import logger from "redux-logger";
import roomsReducer from "./Rooms/rooms";
import reservationsReducer from "./Reservations/Reservations";
import handleUser from "./Authenticate/authentication";

const reducers = combineReducers({
  rooms: roomsReducer,
  reservations: reservationsReducer,
  user: handleUser,
});

const store = createStore(reducers, applyMiddleware(thunk, logger));
export default store;
