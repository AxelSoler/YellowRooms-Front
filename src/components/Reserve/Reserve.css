.reserveRoom {
  height: 85vh;
  color: #fff;
  background-color: #91bd04;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
}

.reserveRoomName {
  width: 70%;
  font-size: 36px;
  border-bottom: 0.01em solid #fff;
  text-align: center;
  padding-bottom: 12px;
}
