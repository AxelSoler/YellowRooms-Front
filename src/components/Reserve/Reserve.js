import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import axios from 'axios';

const Reserve = () => {
  const reservationsURL = 'http://localhost:3001/reservations';
  const rooms = useSelector((state) => state.rooms);
  const [form, setForm] = useState({
    user_id: 0,
    room_id: 0,
    firstName: '',
    start_date: '',
    end_date: '',
    city: '',
  });

  const reserveRoomSubmit = async (e) => {
    e.preventDefault();
    axios.post(reservationsURL, {
      user_id: parseInt(form.user_id, 10),
      room_id: parseInt(form.room_id, 10),
      start_date: form.start_date,
      end_date: form.end_date,
      city: form.city,
    })
      .then((response) => {
        if (response.status === 201) {
          window.location.href = '/Reservations';
        }
      });
  };

  const onChange = (e) => {
    setForm((prevForm) => ({
      ...prevForm,
      [e.target.name]: e.target.value,
    }));
  };

  return (
    <div>
      <h2>Reserve Room</h2>
      <form onSubmit={reserveRoomSubmit}>
        <input onChange={onChange} name="user_id" className="signInput" type="number" placeholder="User" required />
        <select onChange={onChange} name="room_id" required>
          <option>SELECT OPTION</option>
          {rooms.map((room) => (
            <option key={room.id} value={room.id}>{room.name}</option>
          ))}
        </select>
        <input onChange={onChange} name="start_date" className="signInput" type="text" placeholder="Start reservation" required />
        <input onChange={onChange} name="end_date" className="signInput" type="text" placeholder="End reservation" required />
        <input onChange={onChange} name="city" className="signInput" type="text" placeholder="City" required />
        <button disabled={form.room_id === 0 || form.room_id === 'SELECT OPTION'} type="submit">Create Reservation</button>
      </form>
    </div>
  );
};

export default Reserve;
