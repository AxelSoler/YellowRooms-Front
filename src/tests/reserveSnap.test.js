import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Reserve from '../components/Reserve/Reserve';
import store from '../Redux/store';

test('Reserve snapshot', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <Reserve />
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
