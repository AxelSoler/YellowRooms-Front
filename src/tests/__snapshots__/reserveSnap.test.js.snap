// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reserve snapshot 1`] = `
<div>
   No room is set to reserve
</div>
`;
